src/plugins/internalnotification/macros/addType.h

```diff
@@ -0,0 +1,71 @@
+#ifndef TYPE
```

Comment:
Please all-lowercase filenames.

Please add some docu how to use the supermacro.

Will it work for @domhof (#1605) and to replace support.setfuncname in src/tools/gen/template/template.h?

Reply:
While my macros were inspired by his PR I think the only reusable part would be the conversion function. If needed we can extract it but details like error handling would need to be discussed. Theoretically we could have a lib dedicated to string to value and value to string conversions.

Reply:
Yes, let us discuss this.

Reply:
Ok! I'll think about error handling and reply not later than Thursday.

Reply:
> undefined in case of an error.

If possible, the value should stay unchanged.

> What should be name of the conversion library?

If you can implement it as (super)macros no extra library is needed. Otherwise elektra-type might be suitable. We should also add `keySetLong`, `keyGetLong`, etc. functions.

Reply:
> If possible, the value should stay unchanged.

Ok its better to have additional variable declarations & logic in the macro instead of scattering it in the code base.

> If you can implement it as (super)macros no extra library is needed.

Of course! :) A naming macro as input would be required though to avoid name collisions.

> We should also add `keySetLong`, `keyGetLong`, etc. functions.

For such functions a seperate library would be better suited. Also I think this would broaden the scope of the PR too much. For now I'll better stick to the conversion macros.

Reply:
I added a supermacro called `type_create_to_value` for creating "to value" conversion functions in "include/macros". I decided to return the error and have a pointer for the variable because it improves readability:

```C
int variable = 0;
if (!myIntConversion (key, &variable))
{
    // error, variable is still 0
}
```

is better than
```C
int variable = 0;
int error = 0;
variable = myIntConversion (key, error)
if (error)
{
    // error, variable was changed to 0 because of return
}
```

Reply:
>  For such functions a seperate library would be better suited. Also I think this would broaden the scope of the PR too much. For now I'll better stick to the conversion macros.

Yes, of course! I am surprised that you put so much effort in supporting all types. I only mentioned keySetLong to make clear what elektra-type could be used for. But header-only without any lib is the better solution for this case.

Reply:
Ok, great! Then the PR should be completed :)